src/main/java/uk/gov/companieshouse/api/accounts/controller/FilingController.java

```diff
@@ -22,23 +25,46 @@
     private static final Logger LOGGER = LoggerFactory
         .getLogger(CompanyAccountsApplication.APPLICATION_NAME_SPACE);
 
+    private static final String FILING_CONTROLLER_ERROR = "FilingController error:";
+
     @Autowired
     private FilingService filingService;
 
     @GetMapping("/private/transactions/{transactionId}/company-accounts/{companyAccountId}/filings")
     public ResponseEntity generateFiling(@PathParam("transactionId") String transactionId,
         @PathParam("companyAccountId") String accountId, HttpServletRequest request) {
 
+        Transaction transaction =
+            (Transaction) request.getAttribute(AttributeName.TRANSACTION.getValue());
+
+        if (transaction == null) {
+            logRequestError(request, null, "no transaction in request session");
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        CompanyAccountEntity companyAccountEntity =
+            (CompanyAccountEntity) request.getAttribute(AttributeName.COMPANY_ACCOUNT.getValue());
+
+        if (companyAccountEntity == null) {
```

Comment:
The transaction and companyAccountEntity are not used.

Reply:
43107d8